### PR TITLE
Storybook fix

### PR DESCRIPTION
**Pull request by pcvonz**
## This PR is a:

- [ ] New feature
- [ ] Enhancement/Optimization
- [x] Refactor
- [ ] Bugfix
- [ ] Test for existing code
- [ ] Documentation

## Summary

This will add storybook back to Venia

## Additional information

The previous implementation was pretty lackluster (my bad, sorry!). Instead of copy-pasting stuff into the `./storybook/webpack.config.js` from `./webpack.config.js`. This PR actually imports the base webpack config and then adds the necessary configuration to the storybook webpack configuration. 

:rotating_light: This does update `storybook/react`. I think there were build issues with that previously, so just be aware!

fixes: #555

### CODE REVIEW

**Comment by pcvonz** [comment]
> Hey, this looks pretty interesting. Does it solve the `FETCH_ROOT_COMPONENT` issue in your local?

@zetlen Yes, it does resolve the `FETCH_ROOT_COMPONENT` module.

File: packages/venia-concept/package.json [hunk 1]
```diff
@@ -24,14 +24,16 @@
     "start:debug": "node --inspect-brk ./node_modules/.bin/webpack-dev-server --progress --color --env.mode development",
     "test": "jest src",
     "validate-queries": "node ./validate-queries.js",
-    "watch": "webpack-dev-server --progress --color --env.mode development"
+    "watch": "webpack-dev-server --progress --color --env.mode development",
+    "storybook": "start-storybook -p 9001 -c .storybook",
+    "storybook:build": "build-storybook -c .storybook -o storybook-dist"
```
Comment (supernova-at) [inline packages/venia-concept/package.json]: Please keep the alphabetical order of this list 😉

File: packages/venia-concept/package.json [hunk 2]
```diff
@@ -24,14 +24,16 @@
...
   },
   "dependencies": {
     "@magento/eslint-config": "~1.2.3",
     "@magento/peregrine": "^2.0.0-rc.15",
     "@magento/pwa-buildpack": "^2.0.0-rc.15",
     "@magento/upward-js": "^2.0.0-rc.15",
-    "@storybook/react": "~3.4.2",
+    "@storybook/react": "~4.0.11",
```
Comment (supernova-at) [inline packages/venia-concept/package.json]: Last time we attempted this, we ran into `now.sh` issues with Babel and [had to rip it out](https://github.com/magento-research/pwa-studio/pull/540).

@zetlen does the fact that the `now` check completed successfully mean that this PR does not have the same issue?

For reference, we do have [a PR for upgrading to Babel 7](https://github.com/magento-research/pwa-studio/pull/490).

Comment (zetlen) [reply packages/venia-concept/package.json]: @supernova-at I do believe that's what it means!

File: packages/venia-concept/package.json (hunk 1 above)
Comment (zetlen) [reply packages/venia-concept/package.json]: https://github.com/zetlen/eslint-plugin-package-json 👀

**Comment by supernova-at** [comment]
Verification complete, works great! Looks like there are some conflicts to resolve @pcvonz - thanks!

**Pull request merged**